kitsune/search/v2/documents.py

```diff
@@ -91,13 +92,17 @@ def prepare(self, instance):
             'category', 'slug', 'is_archived', 'recent_helpful_votes', 'display_order'
         ]
 
+        obj = cls()
+
+        # Iterate over fields and either set the value directly from the instance
+        # or prepare based on `prepare_<field>` method
         for f in fields:
             try:
-                prepare_method = getattr(self, 'prepare_{}'.format(f))
+                prepare_method = getattr(obj, 'prepare_{}'.format(f))
                 value = prepare_method(instance)
-                setattr(self, f, value)
             except AttributeError:
                 value = getattr(instance, f)
-                setattr(self, f, value)
 
-        self.meta.id = instance.id
+            setattr(obj, f, value)
```

Comment:
I think the `id` should be set with the `document.id`. So it keeps unique and does not create new one while update.